### ISSUE TITLE
JDK 24 new console is causing a serious regression in startup

### ISSUE THREAD

**Issue by geoand** [quarkusio/quarkus#44471]
### Description

When using the `rest-json` quickstart, on my machine I see the following numbers for startup:

* 500ms for JDK 21
* 580ms in 23 
* 650ms in Leyden premain branch (based on JDK 24)

The regressions from 23 to 24 can be reversed by setting `-Djdk.console=java.base`.
We should do that as the first thing in our generated main (unless the user has already set such a property).

P.S. @dmlloyd spotted the JLine issue by looking at the allocation flamegraphs I produced:

21:
![Image](https://github.com/user-attachments/assets/1e6333cc-6447-4e2f-96c3-7e8d18ffffeb)

24:
![Image](https://github.com/user-attachments/assets/d441516f-4eea-4f28-a238-d084204a51ad)

**Comment by geoand** [comment]
@dmlloyd also proposed:

```
we might be able to modify logmanager to automatically set the property before initializing the console
```

which also makes sense

**Comment by dmlloyd** [comment]
I've opened https://issues.redhat.com/browse/LOGMGR-354 with this idea.

**Comment by geoand** [comment]
🙏

**Comment by geoand** [comment]
It turns out that using `-Djdk.console=java.base` gets us back to JDK 21 levels of startup time when using JDK 23, so we probably want to incorporate it sooner rather than later @dmlloyd

**Comment by gsmet** [comment]
Is this with dev mode or plain startup?

I have a vague recollection @Karm opened an issue to determine if the new console could allow us to get rid of Jansi and friends.

**Comment by geoand** [comment]
> Is this with dev mode or plain startup?

All the flamegraphs are from prod mode startup

**Comment by dmlloyd** [comment]
Would you be able/willing to try https://github.com/jboss-logging/jboss-logmanager/pull/491 to see if it fixes the issue?

**Comment by geoand** [comment]
Sure, let me give it a shot

**Comment by geoand** [comment]
That said, the PR itself is totally valid, we just need to do more in Quarkus to make the problem dissapear

**Comment by dmlloyd** [comment]
Since dev mode pertains specifically to the console, maybe we should just set the property there as well. That would avoid the dependency (there's nothing special about the logmanager fix that merits the dependency, it's just setting the property).

**Comment by geoand** [comment]
The class I changed is actually part of the prod jar (which is what I've been testing).

**Comment by dmlloyd** [comment]
Simpler is better! I like it.

**Comment by geoand** [comment]
💪🏽

**Comment by Karm** [comment]
https://github.com/oracle/graal/issues/9884
https://github.com/quarkusio/quarkus/issues/43912
https://github.com/quarkusio/quarkus/issues/43777
https://github.com/oracle/graal/issues/9727

**Comment by geoand** [comment]
Thanks @Karm !

**Comment by geoand** [comment]
@dmlloyd has a new version with your changes been released?

**Comment by dmlloyd** [comment]
No, not yet, as far as I'm aware. @jamezp WDYT about a release?

**Comment by jamezp** [comment]
> No, not yet, as far as I'm aware. @jamezp WDYT about a release?

I'm on PTO today through Monday, but I can do a release on Tuesday.

**Comment by geoand** [comment]
Thanks folks!

**Comment by geoand** [comment]
I opened 44796

**Comment by geoand** [comment]
With the introduction of #44796, to see the difference the console makes you need to take the [rest-json](https://github.com/quarkusio/quarkus-quickstarts/tree/main/rest-json-quickstart) quickstart and start with no flags and with `-Djdk.console=jdk.internal.le` (for the new console)

**Comment by jerboaa** [comment]
@geoand Thanks for the heads-up. I take it's a regression caused by https://bugs.openjdk.org/browse/JDK-8308591? If so the [CSR](https://bugs.openjdk.org/browse/JDK-8309141) has this piece of information:

```
JLine implementation no longer spawns native commands with the fix to JDK-8306983, it is desirable to make the JLine implementation the default Console provider. By lazily instantiating JLine instance, the performance regression can be improved down from >4% (as in JDK-8299137) to 2%.
```

If we have a good and reliable benchmark showing the regression then I'll report this upstream. But probably next year as I don't have the cycles to do it this week. I suppose the claim would be a ~16% regression for this particular quarkus app (is this the dev-console or something else)? That sounds serious enough to warrant a report.

**Comment by geoand** [comment]
Thanks @jerboaa !

The example I mentioned above is a very simple REST application.

The regression we are seeing is in production mode, so there is no dev console.
The reason it shows up in Quarkus is because our logging grabs hold of the console at startup
